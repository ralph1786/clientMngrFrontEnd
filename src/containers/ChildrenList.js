import React, { Component, Fragment } from "react";
import ChildCard from "../components/ChildCard";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { allChildren, searchWord } from "../actions/childActions";
import SearchBar from "../components/UX/SearchBar";
import "./ChildrenList.scss";

export class ChildrenList extends Component {
  componentDidMount() {
    this.props.allChildren();
  }

  changeHandler = e => {
    this.props.search(e.target.value);
  };

  childrenFilteredList = () => {
    const listChildren = this.props.children
      .filter(child =>
        child.name.toLowerCase().includes(this.props.searchTerm.toLowerCase())
      )
      .sort((a, b) => (a.name > b.name ? 1 : -1))
      .map(child => <ChildCard key={child.id} child={child} />);
    return listChildren;
  };

  render() {
    return (
      <Fragment>
        <div className="search-bar">
          <SearchBar
            searchWord={this.props.searchWord}
            onChange={this.changeHandler}
          />
        </div>
        <ul className="children-list">{this.childrenFilteredList()}</ul>
      </Fragment>
    );
  }
}

const mapStateToProps = state => {
  return {
    children: state.children.children,
    searchTerm: state.children.searchWord
  };
};

const mapDispatchToProps = dispatch => {
  return {
    allChildren: () => dispatch(allChildren()),
    search: word => dispatch(searchWord(word))
  };
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(ChildrenList)
);
