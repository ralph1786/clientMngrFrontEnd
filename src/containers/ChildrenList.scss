@import "../utilities/_variables";

.children-list {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(100px, 400px));
  grid-template-rows: repeat(3, 1fr);
  margin-left: 12%;
}

.search-bar input {
  margin-left: 34%;
  width: 30%;
  height: 30px;
  font-size: 1.3em;
  margin-bottom: 7%;
  margin-top: 6%;
  outline: none;
  text-align: center;
  border: 1px solid rgb(68, 67, 67);

  &:focus {
    -webkit-box-shadow: 6px 6px 24px 1px $button-color-hover-opacity;
    -moz-box-shadow: 6px 6px 24px 1px $button-color-hover-opacity;
    box-shadow: 6px 6px 24px 1px $button-color-hover-opacity;
  }
}

@media (max-width: 768px) {
  .children-list {
    margin-left: 25%;
  }
  .search-bar input {
    height: 25px;
    margin-top: 10.5%;
  }
}

@media (max-width: 414px) {
  .children-list {
    margin-left: 15%;
  }
  .search-bar input {
    width: 50%;
    height: 18px;
    margin-left: 25%;
    font-size: 1.1em;
    margin-top: 18%;
  }
}

@media (max-width: 375px) {
  .children-list {
    margin-left: 10%;
  }
}

@media (max-width: 320px) {
  .search-bar input {
    margin-left: 25%;
    height: 16px;
    font-size: 0.9em;
  }
  .children-list {
    margin-left: 4%;
  }
}
