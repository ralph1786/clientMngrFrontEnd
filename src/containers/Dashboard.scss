.dashboard button {
  margin-left: 50px;
  margin-top: 115px;
  padding: 9px;
  border-radius: 25px;
  font-size: 1.04em;
  background-color: rgb(41, 125, 237);
  color: white;
  outline: none;

  &:hover {
    cursor: pointer;
    background-color: rgb(83, 152, 241);
    transition: all 0.4s ease-in;
  }
}
