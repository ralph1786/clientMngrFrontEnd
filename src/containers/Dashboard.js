import React, { Component } from "react";
import ChildrenList from "./ChildrenList";
import "./Dashboard.scss";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";

class Dashboard extends Component {
  render() {
    // console.log(this.props.provider);
    return (
      <div className="dashboard">
        <Link to="/create">
          <button>New Child</button>
        </Link>
        <ChildrenList />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    provider: state.provider
  };
};

export default withRouter(connect(mapStateToProps)(Dashboard));
