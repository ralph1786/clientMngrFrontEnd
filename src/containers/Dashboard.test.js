import React from "react";
import { shallow } from "enzyme";
import { Dashboard } from "./Dashboard";
import { ChildrenList } from "./ChildrenList";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Dashboard provider={{}} setProvider={() => {}} />);
});

it("creates a snapshot for the Dashboard component", () => {
  expect(wrapper).toMatchSnapshot();
});

it("checks that the ChildrenList component is rendered", () => {
  expect(wrapper.find(ChildrenList)).toBeTruthy();
});
