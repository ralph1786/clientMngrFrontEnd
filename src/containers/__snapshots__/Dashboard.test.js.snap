// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a snapshot for the Dashboard component 1`] = `
<div
  className="dashboard"
>
  <withRouter(Connect(ChildrenList)) />
</div>
`;
